### PR TITLE
Remove version-tool dependency from build-tools

### PR DESCRIPTION
**Pull request by curtisman**
Stop keeping track of the `defaultInterdependencyRange` with `MonoRepo` object.  Instead do a look up when it is needed.
This allows the types that is not related to building moved out of `build-tools` and stop having it depending on `version -tools`

### CODE REVIEW

File: build-tools/packages/build-tools/src/common/fluidRepo.ts [hunk 1]
```diff
@@ -321,46 +53,34 @@ export class FluidRepo {
 
 	public readonly packages: Packages;
 
-	public static create(resolvedRoot: string) {
-		const packageManifest = loadFluidBuildConfig(resolvedRoot);
-		return new FluidRepo(resolvedRoot, packageManifest);
-	}
-
-	protected constructor(
+	public constructor(
```
Comment (tylerbutler) [inline build-tools/packages/build-tools/src/common/fluidRepo.ts:56]: why the change to public?

**Review by tylerbutler — COMMENTED** [review]
I need to think more about how this intersects with the changes I want to make to invert the release group/workspace relationship - so we can use a single workspace for the whole repo. I want to get completely out of the business of doing any cross-workspace dependency management and delegate that to the package manager. Some of the first steps are in this PR. Direct link to the config I was playing with: https://github.com/microsoft/FluidFramework/pull/21289/files#diff-9a0994a9d2ddb86f6f1e53fef4e8de22cf87ced6dd81bd0e8866784c2679f450

I've been imagining moving the concept of a release group to be something fluid-build understands, and that sort of implies some changes to how the fluid-build config looks - because it would now be a list of workspaces rather than a list of packages and some folders. Ideally that leads to much simpler package logic overall.

Anyway need to look more deeply and noodle some more.

File: build-tools/packages/build-tools/src/common/fluidRepo.ts (hunk 1 above)
Comment (curtisman) [reply build-tools/packages/build-tools/src/common/fluidRepo.ts:56]: Because we removed the FluidRepo.create and have caller directly do new FluidRepo.

File: build-tools/packages/build-cli/src/args.ts [hunk 2]
```diff
@@ -36,3 +40,38 @@ export const findPackageOrReleaseGroup = (
 		context.independentPackages.find((pkg) => pkg.nameUnscoped === name)
 	);
 };
+
+interface IFluidRepoBuildDir extends IRepoBuildDir {
```
[the remark this answers is not in the document]
Comment (curtisman) [reply build-tools/packages/build-cli/src/args.ts:44]: Without changing the existing config shape, that is very hard to do.   The only thing that might make it not as strange is to duplicate the fields here, so the only dependency is that they are "type-compatible".   And in case things change in the future, we can map the data before passing it into FluidRepo.

Other suggestions welcome.

[the remark this answers is not in the document]
Comment (tylerbutler) [reply build-tools/packages/build-cli/src/args.ts:44]: I think I'm open to changing the config shape somewhat.

[the remark this answers is not in the document]
Comment (curtisman) [reply build-tools/packages/build-cli/src/args.ts:44]: What is the back-compat requirement if we change it?

[the remark this answers is not in the document]
Comment (tylerbutler) [reply build-tools/packages/build-cli/src/args.ts:44]: I thought about this and experimented with completely splitting the fluid-build config from the flub config in #21967. For back-compat, it can still load the old, combined config. I think that it makes sense long-term for the configs to be separate, since there is very little overlap in usage outside of a few places in policy. Curious what you think about it.

**Comment by curtisman** [comment]
Let's go with #21967
